code/game/objects/structures/shower.dm

```diff
@@ -233,18 +234,36 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/shower, (-16))
 		qdel(mist)
 
 
-/obj/machinery/shower/proc/on_entered(datum/source, atom/movable/AM)
+/obj/machinery/shower/proc/on_entered(datum/source, atom/movable/enterer)
 	SIGNAL_HANDLER
+
 	if(actually_on && reagents.total_volume)
-		wash_atom(AM)
+		wash_atom(enterer)
+
+/obj/machinery/shower/proc/on_exited(datum/source, atom/movable/exiter)
+	SIGNAL_HANDLER
+
+	if(!isliving(exiter))
+		return
+
+	var/obj/machinery/shower/locate_new_shower = locate() in get_turf(exiter)
+	if(locate_new_shower && isturf(exiter.loc))
+		return
+	var/mob/living/take_his_status_effect = exiter
+	if(take_his_status_effect.has_status_effect(/datum/status_effect/shower_regen))
+		take_his_status_effect.remove_status_effect(/datum/status_effect/shower_regen)
 
 /obj/machinery/shower/proc/wash_atom(atom/target)
 	target.wash(CLEAN_RAD | CLEAN_WASH)
 	reagents.expose(target, (TOUCH), SHOWER_EXPOSURE_MULTIPLIER * SHOWER_SPRAY_VOLUME / max(reagents.total_volume, SHOWER_SPRAY_VOLUME))
-	if(isliving(target))
-		var/mob/living/living_target = target
-		check_heat(living_target)
-		living_target.add_mood_event("shower", /datum/mood_event/nice_shower)
+	if(!isliving(target))
+		return
+	var/mob/living/living_target = target
+	check_heat(living_target)
+	living_target.add_mood_event("shower", /datum/mood_event/nice_shower)
+	if(living_target.has_status_effect(/datum/status_effect/shower_regen))
+		return
+	living_target.apply_status_effect(/datum/status_effect/shower_regen)
```

Comment:
```suggestion
	living_target.apply_status_effect(/datum/status_effect/shower_regen)
```
`apply_status_effect` will do nothing if they already have it because your status effect has `STATUS_EFFECT_UNIQUE`

Reply:
they thought of everything for me